### PR TITLE
[v11.16.0] Update `next` branch with changes from `main`

### PR DESCRIPTION
**Pull request by laurkim**
Updates `next` branch with latest changes from `main`.
- Resolves conflicts
- Updates renamed components or breaking prop changes ([commit](https://github.com/Shopify/polaris/pull/10382/commits/fb5bc9b872fbd688161ad64bbb8d96feba9d111a))
- Updates any new se23 styles or logic ([commit](https://github.com/Shopify/polaris/pull/10382/commits/e6784cd892481a46d96021cce571a6e21d679452))

> 🚨 Please verify that changes in the last linked commit above look as expected in [storybook](https://5d559397bae39100201eedc1-autktgiyhr.chromatic.com/?path=/story/playground--details-page).

### CODE REVIEW

File: polaris-react/src/components/Avatar/Avatar.scss [hunk 1]
```diff
@@ -19,19 +19,8 @@
...
 
-  #{$se23} & {
-    // stylelint-disable -- se23 overrides
-    --pc-avatar-extra-small-size: 20px;
-    --pc-avatar-small-size: 24px;
-    --pc-avatar-medium-size: 28px;
-    --pc-avatar-large-size: 32px;
-    // stylelint-enable
-    background: var(--p-color-avatar-background-experimental);
-    color: var(--p-color-avatar-color-experimental);
```
Comment (laurkim) [inline polaris-react/src/components/Avatar/Avatar.scss]: These changes are already in the `next` branch from a previous pr to consolidate se23 logic.

**Comment by mrcthms** [comment]
Seeing a little layout shift when going between modes on mobile screens, the correct padding is when you're in the mode where you can see the Save as and Cancel buttons


https://github.com/Shopify/polaris/assets/2562596/5e5a192a-2f40-4e27-ab18-ef9bc40809af

**Comment by laurkim** [comment]
> Seeing a little layout shift when going between modes on mobile screens, the correct padding is when you're in the mode where you can see the Save as and Cancel buttons

Thanks for catching @mrcthms, just pushed up a commit to fix the padding on small screens 👍 

https://github.com/Shopify/polaris/assets/26749317/b126d969-4855-4453-9570-be40124b2d1a

**Review by kyledurand — APPROVED** [review]
🙇

**Review by mrcthms — APPROVED** [review]
Thanks Lo!

**Pull request merged**